torch/_lobpcg.py

```diff
@@ -164,11 +497,43 @@ def lobpcg(A,                   # type: Tensor
 
     """
 
+    # While it is expected that `A` is symmetric,
+    # the `A_grad` might be not. Therefore we perform the trick below,
+    # so that `A_grad` becomes symmetric.
+    # The symmetrization is important for first-order optimization methods,
+    # so that (A - alpha * A_grad) is still a symmetric matrix.
+    # Same holds for `B`.
+    A_sym = (A + A.transpose(-2, -1)) / 2
+    B_sym = (B + B.transpose(-2, -1)) / 2 if (B is not None) else None
+
+    return LOBPCGAutogradFunction.apply(
+        A_sym, k, B_sym, X, n, iK, niter, tol, largest,
+        method, tracker, ortho_iparams, ortho_fparams, ortho_bparams
+    )
+
+
+def _lobpcg(A,                   # type: Tensor
+            k=None,              # type: Optional[int]
+            B=None,              # type: Optional[Tensor]
+            X=None,              # type: Optional[Tensor]
+            n=None,              # type: Optional[int]
+            iK=None,             # type: Optional[Tensor]
+            niter=None,          # type: Optional[int]
+            tol=None,            # type: Optional[float]
+            largest=None,        # type: Optional[bool]
+            method=None,         # type: Optional[str]
+            tracker=None,        # type: Optional[None]
+            ortho_iparams=None,  # type: Optional[Dict[str, int]]
+            ortho_fparams=None,  # type: Optional[Dict[str, float]]
+            ortho_bparams=None,  # type: Optional[Dict[str, bool]]
+            ):
+    # type: (...) -> Tuple[Tensor, Tensor]
+
     if not torch.jit.is_scripting():
         tensor_ops = (A, B, X, iK)
         if (not set(map(type, tensor_ops)).issubset((torch.Tensor, type(None))) and has_torch_function(tensor_ops)):
             return handle_torch_function(
-                lobpcg, tensor_ops, A, k=k,
+                _lobpcg, tensor_ops, A, k=k,
```

Comment:
I think this whole block of code should be moved to the new `torch.lobpcg()` function. It needs to run "above" the custom Function.

Reply:
I actually did it, but stashed and pushed the wrong copy ;)